programs/solana/programs/ics07-tendermint/src/instructions/update_client.rs

```diff
@@ -197,71 +202,910 @@ fn handle_consensus_state_storage<'info>(
+        pub new_consensus_state_pda: Pubkey,
+        pub payer: Pubkey,
+        pub instruction: Instruction,
+        pub accounts: Vec<(Pubkey, Account)>,
+    }
+
+    pub struct HappyPathTestScenario {
+        pub client_state_pda: Pubkey,
+        pub trusted_consensus_state_pda: Pubkey,
+        pub new_consensus_state_pda: Pubkey,
+        pub payer: Pubkey,
+        pub instruction: Instruction,
+        pub accounts: Vec<(Pubkey, Account)>,
+        pub update_message_fixture: UpdateClientMessageFixture,
+    }
+
+    fn create_clock_account(unix_timestamp: i64) -> (Pubkey, Account) {
+        let clock = Clock {
+            slot: 1000,
+            epoch_start_timestamp: 0,
+            epoch: 1,
+            leader_schedule_epoch: 1,
+            unix_timestamp,
+        };
+
+        // Serialize the Clock struct using bincode
+        let data = bincode::serialize(&clock).expect("Failed to serialize Clock");
+
+        (
+            CLOCK_ID,
+            Account {
+                lamports: 1,
+                data,
+                owner: solana_sdk::sysvar::ID,
+                executable: false,
+                rent_epoch: 0,
+            },
+        )
+    }
+
+    fn create_update_client_instruction(
+        client_state_pda: Pubkey,
+        trusted_consensus_state_pda: Pubkey,
+        new_consensus_state_pda: Pubkey,
+        payer: Pubkey,
+        client_message: Vec<u8>,
+    ) -> Instruction {
+        let update_msg = UpdateClientMsg { client_message };
+        let instruction_data = crate::instruction::UpdateClient { msg: update_msg };
+
+        Instruction {
+            program_id: crate::ID,
+            accounts: vec![
+                AccountMeta::new(client_state_pda, false),
+                AccountMeta::new_readonly(trusted_consensus_state_pda, false),
+                AccountMeta::new(new_consensus_state_pda, false),
+                AccountMeta::new(payer, true),
+                AccountMeta::new_readonly(system_program::ID, false),
+            ],
+            data: instruction_data.data(),
+        }
+    }
+
+    fn create_empty_consensus_state_account() -> Account {
+        Account {
+            lamports: 0,
+            data: vec![],
+            owner: system_program::ID,
+            executable: false,
+            rent_epoch: 0,
+        }
+    }
+
+    fn setup_test_accounts_with_new_consensus_state(
+        initialized_accounts: Vec<(Pubkey, Account)>,
+        new_consensus_state_pda: Pubkey,
+        payer: Pubkey,
+        payer_lamports: u64,
+    ) -> Vec<(Pubkey, Account)> {
+        let mut accounts = initialized_accounts;
+        accounts.push((
+            new_consensus_state_pda,
+            create_empty_consensus_state_account(),
+        ));
+
+        // Update payer lamports
+        for (pubkey, account) in accounts.iter_mut() {
+            if pubkey == &payer {
+                account.lamports = payer_lamports;
+                break;
+            }
+        }
```

Comment:
I haven't checked that, but would it be possible to write it in a shorter form?

```suggestion
        if let Some((_, account)) = accounts.iter_mut().find(|(key, _)| *key == payer) {
            account.lamports = payer_lamports;
        }
```